cargo-near/src/commands/deploy/mod.rs

```diff
@@ -0,0 +1,147 @@
+#[interactive_clap(input_context = near_cli_rs::GlobalContext)]
+#[interactive_clap(output_context = ContractContext)]
+#[interactive_clap(skip_default_from_cli)]
+pub struct Contract {
+    #[interactive_clap(skip_default_input_arg)]
+    #[interactive_clap(flatten)]
+    /// Specify a build command args:
+    build_command_args: build_command::BuildCommand,
+    #[interactive_clap(skip_default_input_arg)]
+    /// What is the contract account ID?
+    contract_account_id: near_cli_rs::types::account_id::AccountId,
+    #[allow(dead_code)]
+    #[interactive_clap(skip)]
+    file_path: crate::types::utf8_path_buf::Utf8PathBufInner,
+    #[interactive_clap(subcommand)]
+    initialize: InitializeMode,
+}
+
+#[derive(Debug, Clone)]
+pub struct ContractContext(near_cli_rs::commands::contract::deploy::ContractFileContext);
+
+impl ContractContext {
+    pub fn from_previous_context(
+        previous_context: near_cli_rs::GlobalContext,
+        scope: &<Contract as interactive_clap::ToInteractiveClapContextScope>::InteractiveClapContextScope,
+    ) -> color_eyre::eyre::Result<Self> {
+        Ok(Self(
+            near_cli_rs::commands::contract::deploy::ContractFileContext {
+                global_context: previous_context,
+                receiver_account_id: scope.contract_account_id.clone().into(),
+                signer_account_id: scope.contract_account_id.clone().into(),
+                code: scope.file_path.read_bytes()?,
+            },
+        ))
+    }
+}
+
+impl From<ContractContext> for near_cli_rs::commands::contract::deploy::ContractFileContext {
+    fn from(item: ContractContext) -> Self {
+        item.0
+    }
+}
+
+impl interactive_clap::FromCli for Contract {
+    type FromCliContext = near_cli_rs::GlobalContext;
+    type FromCliError = color_eyre::eyre::Error;
+    fn from_cli(
+        optional_clap_variant: Option<<Self as interactive_clap::ToCli>::CliVariant>,
+        context: Self::FromCliContext,
+    ) -> interactive_clap::ResultFromCli<
+        <Self as interactive_clap::ToCli>::CliVariant,
+        Self::FromCliError,
+    >
+    where
+        Self: Sized + interactive_clap::ToCli,
+    {
+        let mut clap_variant = optional_clap_variant.unwrap_or_default();
+
+        let build_command_args =
+            if let Some(cli_build_command_args) = &clap_variant.build_command_args {
+                build_command::BuildCommand {
+                    release: cli_build_command_args.release,
+                    embed_abi: cli_build_command_args.embed_abi,
+                    doc: cli_build_command_args.doc,
+                    no_abi: cli_build_command_args.no_abi,
+                    out_dir: cli_build_command_args.out_dir.clone(),
+                    manifest_path: cli_build_command_args.manifest_path.clone(),
+                    color: cli_build_command_args.color.clone(),
+                }
+            } else {
+                build_command::BuildCommand::default()
+            };
```

Comment:
It seems that something like this should work:

```suggestion
        let build_command_args = clap_variant.build_command_args.map(BuildCommand::from_cli).unwrap_or_default();
```